src/main/java/org/folio/circulation/services/EventPublisher.java

```diff
@@ -174,11 +212,50 @@ public EventPublisher(PubSubPublishingService pubSubPublishingService) {
     return completedFuture(succeeded(requestAndRelatedRecords));
   }
 
-  public CompletableFuture<Result<Void>> publishLogRecord(JsonObject payload, LogEventPayloadType payloadType) {
+  public CompletableFuture<Result<Void>> publishAgedToLostEvent(Loan loan) {
+    return publishLogRecord(LoanLogContext.from(loan)
+      .withDescription(String.format("Due date: %s", loan.getAgedToLostDateTime())).asJson(), LOAN);
+  }
+
+  public CompletableFuture<Result<Void>> publishClosedLoanEvent(Loan loan) {
+    if (CHECKED_IN.getValue().equalsIgnoreCase(loan.getAction())) {
+      return publishLogRecord(LoanLogContext.from(loan)
+        .withAction(LogContextActionResolver.resolveAction(CLOSED_LOAN.getValue()))
+        .withServicePointId(loan.getCheckInServicePointId()).asJson(), LOAN);
+    }
+    return publishLogRecord(LoanLogContext.from(loan)
+      .withServicePointId(loan.getCheckoutServicePointId()).asJson(), LOAN);
+  }
+
+  public CompletableFuture<Result<Loan>> publishMarkedAsMissingLoanEvent(Loan loan) {
+    return publishLogRecord(LoanLogContext.from(loan)
+      .withDescription(String.format("Additional information: %s", loan.getActionComment())).asJson(), LOAN)
+      .thenApply(r -> r.map(v -> loan));
+  }
+
+  public CompletableFuture<Result<LoanAnonymizationRecords>> publishAnonymizeEvents(LoanAnonymizationRecords records, Clients clients) {
+    LoanRepository loanRepository = new LoanRepository(clients);
+    return loanRepository.findByIds(records.getAnonymizedLoans())
```

Comment:
Is it possible to pass collection of loans instead of rereading them from storage?